### PR TITLE
Convert some Lua hooks to the new style to add documentation

### PR DESCRIPTION
**Pull request by asarium**
This takes a few of the existing hooks and converts them to the new
style of hook definition which can include more extensive documentation
and enumeration of supported hook variables.

### CODE REVIEW

File: code/object/collideshipship.cpp [hunk 1]
```diff
@@ -1357,18 +1362,21 @@ int collide_ship_ship( obj_pair * pair )
...
+				scripting::hooks::OnShipCollision->run(scripting::hook_param_list(scripting::hook_param("Self", 'o', A),
+														   scripting::hook_param("Object", 'o', B),
+														   scripting::hook_param("Ship", 'o', A),
+														   scripting::hook_param("ShipB", 'o', B),
+														   scripting::hook_param("Hitpos", 'o', world_hit_pos)),
+					A);
 			}
 			if((b_override && !a_override) || (!b_override && !a_override))
 			{
-				// Yes, this should be reversed.
```
Comment (Goober5000) [inline code/object/collideshipship.cpp:1367]: I would leave the comment in the code, or even make it more explicit that the swapping of A and B is intentional.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Added a note about a comment.

Also, if you're going to be messing with the `On Death` hook anyway, I suggest deprecating that hook and making new individual `On Ship Death` and `On Asteroid Death` hooks.

File: code/object/collideshipship.cpp (hunk 1 above)
Comment (asarium) [reply code/object/collideshipship.cpp:1367]: Good point. Added the comment again.

**Comment by asarium** [comment]
Also a good point about the deprecation. However, that will require some more work to integrate it into the existing deprecation system so that is out of scope for this PR.

**Review by Goober5000 — APPROVED** [review]
LGTM

**Comment by TRBlount** [comment]
Is this still in Draft status?

**Comment by z64555** [comment]
Deferring to 21.6 due to lack of feedback.

**Comment by TRBlount** [comment]
If this is still in Draft status, then it will be delayed until 22.2

**Comment by z64555** [comment]
Closing upon request due to lack of progress.  Please resubmit when you get back to it.

**Pull request closed without merging**